Foundation/NSURLResponse.swift

```diff
@@ -154,28 +166,192 @@ public class NSHTTPURLResponse : NSURLResponse {
         @abstract Returns the HTTP status code of the receiver. 
         @result The HTTP status code of the receiver. 
     */
-    public var statusCode: Int { NSUnimplemented() }
+    public let statusCode: Int
     
-    /*! 
-        @method allHeaderFields
-        @abstract Returns a dictionary containing all the HTTP header fields
-        of the receiver.
-        @discussion By examining this header dictionary, clients can see
-        the "raw" header information which was reported to the protocol
-        implementation by the HTTP server. This may be of use to
-        sophisticated or special-purpose HTTP clients.
-        @result A dictionary containing all the HTTP header fields of the
-        receiver.
-    */
-    public var allHeaderFields: [NSObject : AnyObject] { NSUnimplemented() }
+    /// Returns a dictionary containing all the HTTP header fields
+    /// of the receiver.
+    ///
+    /// By examining this header dictionary, clients can see
+    /// the "raw" header information which was reported to the protocol
+    /// implementation by the HTTP server. This may be of use to
+    /// sophisticated or special-purpose HTTP clients.
+    ///
+    /// - Returns: A dictionary containing all the HTTP header fields of the
+    /// receiver.
+    ///
+    /// - Important: This is an *experimental* change from the
+    /// `[NSObject: AnyObject]` type that Darwin Foundation uses.
+    public let allHeaderFields: [String: String]
```

Comment:
Also note that this has to handle the case where the same header is provided with different capitalizations to the initializer (in that event, the resulting `allHeaderFields` dictionary should only contain one of the equivalent keys and its corresponding value, and the conflicts should be discarded).